sync_test.go

```diff
@@ -0,0 +1,207 @@
+package legs
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	leveldb "github.com/ipfs/go-ds-leveldb"
```

Comment:
can we use a mem db since this is only for tests and lets us avoid otherwise pulling in the dependency?

Reply:
Do we have any memory datastore that is "concurrent-friendly"? (I don't know of any). I had to use levelDB parallel read/writes in the MapDatastore for one of the tests (is the only one I knew).

Reply:
That might be true.

is it easy to switch this file to be in a `legs_test` package?

Reply:
I don't think we can without exposing `latestSync` through the `Subscriber` interface. One thing I can probably do is to create a new `MapDatastore` in each step of the test and load the part of the dag already synced to avoid having to use LevelDB. It is a bit more of work, but I'll look into this.

Reply:
It's not the end of the world, just hoping to not pull it into the dependency tree unnecessarily

Reply:
No, you are right. It's not a great idea to pull LevelDB just for testing when users will actually end-up hooking their preferred datastore when using the library. I'll look into it tomorrow.